sdks/go/README.md

```diff
@@ -84,56 +145,237 @@ sentence: 1
+
+You **do not** need to know or care about Java or Python to use the Go SDK for your data processing goals.
+
+Java is **only** required to execute any [gradle](https://gradle.org/) commands configured in the
+[Beam](https://github.com/apache/beam) repository. It will be obvious whether you will execute gradle commands in
+sections below. You **do not** need to install [gradle](https://gradle.org/) and simply use the enclosed
+`$BEAM_ROOT/gradlew` executable available at the root of the [Beam](https://github.com/apache/beam) repository.
+
+Python is **only** required if you need to validate tests against the
+[Portable Python Runner](#portable-python-runner). If you are not testing against this
+runner, ignore the Python requirement.
+
+#### 2. Docker
+
+[Docker](https://www.docker.com/) is required for certain but not all runners [See definition](#runner).
+
+As of this writing, [colima](https://github.com/abiosoft/colima), a preferred docker alternative for some developers,
+did not work.
+
+#### 3. Flock
+
+[sdks/go/run_with_go_version.sh](run_with_go_version.sh) requires the use of
+[flock](https://github.com/discoteq/flock).
+
+### Execution
+
+#### 1. Navigate to the go.mod directory
+
+Open a terminal and navigate into the go.mod containing directory of the Beam repository. (See above for what
+`$BEAM_ROOT` means). Notice that you are entering the `$BEAM_ROOT/sdks` and not `$BEAM_ROOT/sdks/go`.
+
+```sh
+cd $BEAM_ROOT/sdks
+```
+
+#### 2. Run Go test
+
+Run go test as you would any Go project.
 
+For unit tests in the exported `pkg/beam` package:
 ```
-$  go run wordcount.go --runner=dataflow --project=<YOUR_GCP_PROJECT> --region=<YOUR_GCP_REGION> --staging_location=<YOUR_GCS_LOCATION>/staging --worker_harness_container_image=<YOUR_SDK_HARNESS_IMAGE_LOCATION> --output=<YOUR_GCS_LOCATION>/output
+go test ./go/pkg/beam...
 ```
 
-The output is a GCS file in this case:
+For integration, load, and regression tests:
+```
+go test ./go/test/...
+```
+
+### Runner validations
+
+The following documents various [Runner](#runner) validation tests related to test execution of the Go SDK
+**in this repository** (in contrast to your own Go SDK dependent repository and projects).
+You'll see some documentation that deviates from Go test execution convention to run gradle commands.
+
+As a Go developer, you might ask yourself, "Why gradle instead of 'go test' 😬?"
+Under the hood, various [Runner](#runner) architecture involves resources and non-Go languages that Go SDK runner
+validations require. The cost of deviating from the `go test` convention, comes with the benefit of allowing you
+as a Beam contributor to focus on the Go SDK and ignore the underlying architecture specifics. The gradle commands
+conveniently automate and encapsulate these details.
+
+#### 1. Navigate to the Beam root directory
+
+Open a terminal and navigate into the root directory of the Beam repository. (See [Beam Repository](#beam-repository)
+for what `$BEAM_ROOT` means).
```

Comment:
```suggestion
Open a terminal and navigate into the root directory of the Beam repository. 
```

Linking the section explaining $BEAM_ROOT feels unnecessary since you're explained where they're navigating here in plain-text already. You can arguably drop the `cd` command block too